src/binarytree.jl

```diff
@@ -143,3 +143,67 @@ function _printkeyvalue(io::IO, node::BinaryNode)
     show(ioctx, val)
   end
 end
+
+# -----------
+#  UTILITIES
+# -----------
+
+minnode(tree::BinaryTree) = minnode(root(tree))
+
+function minnode(node::BinaryNode)
+  leftnode = left(node)
+  isnothing(leftnode) ? node : minnode(leftnode)
+end
+
+minnode(node::Nothing) = nothing
+
+maxnode(tree::BinaryTree) = maxnode(root(tree))
+
+function maxnode(node::BinaryNode)
+  rightnode = right(node)
+  isnothing(rightnode) ? node : maxnode(rightnode)
+end
+
+maxnode(node::Nothing) = nothing
+
+function abovebelow(tree::BinaryNode, x::BinaryNode)
```

Comment:
I think the second argument should be the key. In Meshes PR, only the key is needed, like this:
```julia
function abovebelow(tree::BinaryTree, k)
  # code
end
```
The Meshes PR can be refactored like this:
```julia
BinaryTrees.abovebelow(𝒯, s)
```

Reply:
In terms of terminology, what do you think of renaming this function to `prevnext` to emphasize the underlying ordering? The name `abovebelow` seems to be slightly tanget compared to prev/next or predecessor/successor.

Reply:
I agree.

Reply:
I agree.